IONFileViewerLib/IONFLVWManager.swift

```diff
@@ -1,3 +1,110 @@
+        guard file.lastPathComponent.contains(".") else { throw IONFLVWError.missingFileExtension }
+        guard fileManager.fileExists(atPath: file.path) else { throw IONFLVWError.fileDoesNotExist(atPath: filePath) }
+        
+        try openDocumentFromLocalPath(file, completion)
+    }
+    
+    public func openDocumentFromResources(assetPath: String, completion: @escaping (() -> Void)) throws {
+        let url = try getResourceURL(assetPath)
+        
+        try openDocumentFromLocalPath(url, completion)
+    }
+    
+    public func openDocumentFromUrl(url: String, completion: @escaping ((IONFLVWError?) -> Void)) throws {
+        guard !url.isEmpty else { throw IONFLVWError.invalidEmptyURL }
+        guard url.isValidUrl(), let fileUrl = URL(string: url.replacingOccurrences(of: " ", with: "%20")) else {
+            throw IONFLVWError.invalidURL(forUrl: url)
+        }
+        
+        if let viewController = rootViewController {
+            let fileViewerOpenDocument = IONFLVWDocumentOpener(viewController: viewController)
+            fileViewerOpenDocument.openDocumentFromUrl(url: fileUrl, completion: { (inner: IONFLVWCompletionHandler) in
+                do {
+                    try inner()
+                    completion(nil)
+                } catch {
+                    completion((error as? IONFLVWError) ?? .couldNotOpenDocument)
+                }
+            })
+        }
+    }
+    
+    private func openDocumentFromLocalPath(_ url: URL, _ completion: @escaping (() -> Void)) throws {
+        if let viewController = rootViewController {
+            let fileViewerOpenDocument = IONFLVWDocumentOpener(viewController: viewController)
+            try fileViewerOpenDocument.openDocumentFromLocalPath(filePath: url) {
+                completion()
+            }
+        }
+    }
+}
+
+extension IONFLVWManager: IONFLVWPreviewMediaManager {
+    public func previewMediaContentFromLocalPath(filePath: String) throws {
+        guard !filePath.isEmpty else { throw IONFLVWError.emptyFilePath }
+        guard let file = URL(string: filePath) else { throw IONFLVWError.couldNotOpenDocument }
+        guard fileManager.fileExists(atPath: file.path) else { throw IONFLVWError.fileDoesNotExist(atPath: filePath) }
+        
+        try previewMediaContent(file)
+    }
+    
+    public func previewMediaContentFromResources(assetPath: String) throws {
+        let url = try getResourceURL(assetPath)
+        
+        try previewMediaContent(url)
+    }
+    
+    public func previewMediaContentFromUrl(url: String) throws {
+        guard !url.isEmpty else { throw IONFLVWError.invalidEmptyURL }
+        guard url.isValidUrl(), let fileUrl = URL(string: url) else { throw IONFLVWError.invalidURL(forUrl: url) }
+        
+        try previewMediaContent(fileUrl)
+    }
+    
+    private func previewMediaContent(_ url: URL) throws {
+        if let viewController = rootViewController {
```

Comment:
Same comment as the ones above, we could possibly use `guard let` and return early.